### PR TITLE
Move changes from #572 into foxy

### PR DESCRIPTION
**Pull request by nkalupahana**
The fix from PR #572 is essential to get RViz to work on macOS. Otherwise, crashing RViz is as simple as this:

- Open RViz
- Add /goal_pose to the display
- Uncheck Pose in the left sidebar

The `foxy` branch is what's used for the binary build of ROS2 Foxy on macOS, so this change needs to be in the foxy branch in order for rviz to be usable on macOS. This PR moves that change.

### CODE REVIEW

**Comment by Ace314159** [comment]
Can anyone look at this? This would be really helpful for us and all ROS2 macOS users as well.

**Comment by nkalupahana** [comment]
@clalancette or anyone else, can we get this merged in? It's just a one line change, and it'll make a big difference for anyone using rviz on mac.

**Review by clalancette — APPROVED** [review]

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14800/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9549/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12466/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14951/)

**Comment by nkalupahana** [comment]
@clalancette is the CI an issue, or is this good to go? This change is in the `ros2` branch as well, so I can't imagine that it would cause issues.

**Comment by clalancette** [comment]
I think all of those warnings are well-known on Foxy.  But pinging @jacobperron for approval before merging this.

**Comment by nkalupahana** [comment]
@jacobperron please approve when you get the chance!

**Review by jacobperron — APPROVED** [review]
Apologies for the delay, LGTM!

**Pull request merged**
